app/src/main/java/com/example/una/models/Charity.java

```diff
@@ -0,0 +1,47 @@
+package com.example.una.models;
+
+import android.location.Address;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Locale;
+
+public class Charity {
+
+    // attributes
+    public String ein; // federal Employer Identification Number
+    public String name;
+    public String description; // tag line describing the charity
+    public String mission; // organization's mission statement
+    public String category;
+    public String cause;
+    public Address donationAddress;
+    public boolean isFeatured;
+    // TODO - add donorList, photoUrl, verified
+
+    // deserialize the JSON
+    public static Charity fromJSON(JSONObject jsonObject) throws JSONException {
+        Charity charity = new Charity();
+
+        // extract the values from JSON
+        charity.ein = jsonObject.getString("ein");
+        charity.name = jsonObject.getString("charityName");
+        charity.description = jsonObject.getString("tagLine");
+        charity.mission = jsonObject.getString("mission");
+        charity.category = jsonObject.getString("categoryName");
+        charity.cause = jsonObject.getString("causeName");
+        charity.isFeatured = false;
+
+        // set charity location
+        charity.donationAddress = new Address(Locale.ENGLISH);
+        charity.donationAddress.setAddressLine(0, jsonObject.getString("streetAddress1"));
```

Comment:
I don't think this is accurate.

Looking at the doc, seems like they have two addresses `mailingAddress` and `donationAddress`.
Each of that field has the details. Calling `jsonObject.getString("streetAddress1")` on the outer object of Charity will return `null`

Reply:
Would `charity.donationAddress.setAddressLine(0, jsonObject.getJSONObject("donationAddress").getString("streetAddress1"));` for line 38 be correct?

Reply:
Yes, if donation address is what you want to store here. You might want to consider the case when they don't have a `donationAddress` set in the server